bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -205,21 +204,41 @@ private void checkConnection() throws CoolMasterClientError {
                 }
 
                 InputStream in = localSocket.getInputStream();
-                /* Flush anything pending in the input stream */
-                while (in.available() > 0) {
-                    in.read();
+
+                // Sink (clear) buffer until earlier of the SINK_TIMEOUT or > prompt
+                try {
+                    localSocket.setSoTimeout(SINK_TIMEOUT);
+                    while (true) {
+                        int b = in.read();
+                        if (b == -1) {
+                          break;
+                        }
+                        if (b == PROMPT) {
+                            if (in.available() > 0) {
+                                throw new IOException("Unexpected data following prompt");
+                            }
+                            logger.trace("Buffer empty following unsolicited > prompt");
+                            return;
+                        }
+                    }
+                } catch (final IOException ignoredAsLikelySocketTimeout) {
+                } finally {
```

Comment:
please log this anyway, you can log it as DEBUG

Reply:
I narrowed the exception instead to. That way it reflects the expected blue sky case (which is `in.read()` timing out as the buffer is now empty). Other `IOException` cases are already caught and logged in the enclosing `try` statement.